### DISCUSSION TITLE
[Question] Availability of environments in the paper

### DISCUSSION THREAD

**Discussion by VincentYu68** [isaac-sim/IsaacLab#11] [General]
Hi, there seems to be several environments missing in the repo right now compared to the paper. I'm wondering if they are not in the orbit_env directory, or they are yet to be released? If latter, is there any timeline on when they'll be available?

Thanks!

**Comment by animesh-garg** [comment]
Thanks for the note.

Many of the environments in the paper are listed under the "Features" section of the [website](https://isaac-orbit.github.io/orbit/source/features/environments.html).

The mentioned rigid body environments and legged robot environments will be released shortly. However, for deformable body manipulation environments, we decided that providing the CPU implementations for these tasks is not substantially beneficial since their throughput is limited and there might be deprecation due to API changes from the simulator side. 

The release of cloth manipulation environments will be happening soon. We already have some promising throughput results for them with the tensor API of cloth available in Isaac Sim 2022.2.0.

We are working on the tensorized GPU APIs for them which should be made available in the coming releases of Isaac Sim. It is hard to pin down the exact timeline for all of them at the moment since some of these feature requests go beyond our team.
An estimated development roadmap for the release of all the environments can be found [here](https://isaac-orbit.github.io/orbit/source/refs/roadmap.html).

We thank you for your patience.

**Reply by LuukvBree** [reply]
Hi, 

I'm curious about the development roadmap you mentioned, but the link isn't working for me. I'm currently exploring AMP integration, which I understand isn't available in Orbit yet, is that correct? Could you provide more details on how AMP might be incorporated into Orbit in the future? Alternatively, could you direct me to where I can find the roadmap for this?

Kind regards

**Reply by eferreirafilho** [reply]
Hi @animesh-garg , thanks for the great work!


Is there any up-to-date roadmap for new envs code? I'm especially interested in deformable manipulation in IsaacLab. Thanks!

**Comment by sirkrisp** [comment]
The official documentation website is still not listing many of the manipulation environments featured in the paper (in particular manipulation of clothes and liquids). Could you give an update on the current progress? (link to roadmap is broken) Thanks!

**Reply by Mayankm96** [reply]
Hi,

Unfortunately, we had to take our development roadmap down since sticking to the plan mentioned there became challenging. Our priorities frequently shift to accommodate both customer needs and internal research demands. Nonetheless, we want to assure you that the creation of manipulation environments remains our top priority. 

We will be adding some of them soon in our upcoming v0.3 release. A more formal announcement will follow in the coming week (hopefully).

**Comment by duidui6666** [comment]
I would like to ask if the official document or website will update the introduction of the task of opening the cabinet by the mobile manipulator or the task of opening the cabinet by the manipulator

**Reply by Mayankm96** [reply]
Hi, we do have a manipulator opening a cabinet task: https://github.com/NVIDIA-Omniverse/orbit/blob/main/source/extensions/omni.isaac.orbit_tasks/omni/isaac/orbit_tasks/manipulation/cabinet/config/franka/joint_pos_env_cfg.py

We are constantly adding new environments. Please check the website: https://isaac-orbit.github.io/orbit/source/features/environments.html

**Reply by duidui6666** [reply]
Oh, yes, I just found it. Is there any update on the task of moving the robot to open the cabinet?

**Comment by Mayankm96** [comment]
It has been difficult to balance re-adding these environments and improving the framework with other requests. We are actively working on both fronts and hope to reach a state where all the environments exist and work out of the box. However, some of the environments are held back since we are waiting for Isaac Sim to improve itself (for instance, FEM simulation for clothes).

Since this has been a repeated topic of discussion on GitHub, we have made a single issue to keep track of our progress: https://github.com/isaac-sim/IsaacLab/issues/748. I hope this helps!
